File: 2023-2024/sourcecode/mod2/lab9/SharedPreferencesBasicsJava/app/src/main/java/com/example/sharedpreferencesbasicsjava/repository/UserRepository.java
package com.example.sharedpreferencesbasicsjava.repository;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class UserRepository {

    private final SharedPreferences sharedPref;
    private final MutableLiveData<String> _username = new MutableLiveData<>();

    public UserRepository(Application application) {
        sharedPref = application.getSharedPreferences("fileName", Context.MODE_PRIVATE);
        _username.setValue(sharedPref.getString("username", ""));
    }

    public LiveData<String> getUsername() {
        return _username;
    }

    public void add(String newUsername) {
        SharedPreferences.Editor edit = sharedPref.edit();
        edit.putString("username", newUsername).apply();
        _username.setValue(newUsername);
    }

    public void clear() {
        SharedPreferences.Editor edit = sharedPref.edit();
        edit.putString("username", "").apply();
        _username.setValue("");
    }
}

File: 2023-2024/sourcecode/mod2/lab9/SharedPreferencesBasicsKotlin/app/src/main/java/com/example/sharedpreferencesbasicskotlin/repository/UserRepository.kt
package com.example.sharedpreferencesbasicskotlin.repository

import android.app.Application
import android.content.Context.MODE_PRIVATE

class UserRepository(application: Application) {

    private val sharedPref = application.getSharedPreferences("fileName", MODE_PRIVATE)

    private var _username: String = sharedPref.getString("username", "") ?: ""
    val username: String
        get() = _username

    fun add(newUsername: String) {
        val edit = sharedPref.edit()
        edit.putString("username", newUsername).apply()
        _username = newUsername
    }

    fun clear() {
        val edit = sharedPref.edit()
        edit.putString("username", "").apply()
        _username = ""
    }
}